### PR TITLE
Merge feat/account-page -> wip/web-rework

### PR DESCRIPTION
**Pull request by ceriddenn**
Changes include:
+ LoginTabItems Component (not done will be introduced in v1.1 ig)
+ DeleteAccModal Component (Modal that confirms account deletion and runs the logic)
+ AccGameplayItems Component (Shows user gameplay items (right side of account page))
+ AccGameplayItemExtended Component (Extension of AccGameplayItems and runs and renders the logic for contacting the youtube api to gather the title and other youtube video data we may need)
+ Account page (shows linked accounts and renders the main components)
Update: Api Helper for api requests
Update: Navbar (added new items to the navbar and design highlighting for the current page)
ENDPOINTS:
+ unlinkAccount (unlinks an account.. duh)
+ getLinkedAccounts (find accounts linked to the current user's session)
Updated deleteUser and schema.prisma so that deleting a user document deletes all of the child document relations to the user using cascade option)
Updated: additional.d.ts for env types for youtube_api_key
PS: A youtube api key is now required (will be easy to get as it can be created on the same page as google oauth. The youtube data api v3 needs to be enabled though as it is disabled by default)

### CODE REVIEW

File: apps/web/.env.example [hunk 1]
```diff
@@ -7,13 +7,15 @@ GOOGLE_CLIENT_SECRET=example
...
+NEXTAUTH_URL="example"
+//redis 
+REDIS_HOST="example"
+REDIS_PASSWORD="example"
+REDIS_PORT=example
+//ratelimit
+// below is an example for 1 requests every 5 mins
+RATELIMIT_MAX_REQUESTS=400
+// time in ms below
+RATELIMIT_TIME=3
```
Comment (Huskydog9988) [inline apps/web/.env.example:21]: breaks redis connection
```suggestion
# NextAuth options https://next-auth.js.org/configuration/options
NEXTAUTH_URL="example"

# Cockroach connection url
DATABASE_URL="example"

# Youtube api key
YOUTUBE_API_KEY="example"

# redis
REDIS_URL=example

# ratelimit
# below is an example for 1 requests every 5 mins
RATELIMIT_MAX_REQUESTS=1
# time in ms below
RATELIMIT_TIME=300000
```

File: apps/web/additional.d.ts [hunk 2]
```diff
@@ -4,11 +4,15 @@ declare namespace NodeJS {
...
     RATELIMIT_MAX_REQUESTS: number;
     RATELIMIT_DURATION: number;
+    YOUTUBE_API_KEY: string;
   }
 }
 declare namespace NodeJS {
   interface ProcessEnv {
     DISCORD_CLIENT_SECRET: string;
-    REDIS_URL: string;
+    REDIS_PORT: number;
```
Comment (Huskydog9988) [inline apps/web/additional.d.ts:13]:
```suggestion
    REDIS_URL: string;
```

File: apps/web/components/AccGameplayItemExtended.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { trpc } from '@utils/trpc';
+import Loading from './Loading';
+import { BiTrash } from 'react-icons/bi';
+import { getYtVidDataFromId } from '@utils/helpers/apiHelper';
+// TODO
+interface Item {}
+
+interface Id {
+  id: string;
+}
```
Comment (Huskydog9988) [inline apps/web/components/AccGameplayItemExtended.tsx:22]: this is never used
```suggestion
```

File: apps/web/components/Navbar.tsx [hunk 4]
```diff
@@ -179,4 +180,14 @@ const NAV_ITEMS: Array<NavItem> = [
...
+  {
+    label: 'Account',
+    href: '/account',
+    pathName: '/account',
+  },
+  {
+    label: 'Login',
+    href: '/auth/login',
+    pathName: '/auth/login',
+  },
```
Comment (Huskydog9988) [inline apps/web/components/Navbar.tsx:192]: shouldn't this be conditional?

File: apps/web/next.config.js [hunk 5]
```diff
@@ -14,6 +14,9 @@ const nextConfig = {
   images: {
     domains: ['cdn.discordapp.com'],
   },
+  env: {
+    YOUTUBE_API_KEY: process.env.YOUTUBE_API_KEY,
+  },
```
Comment (Huskydog9988) [inline apps/web/next.config.js:19]: part of the yt util discussion

File: apps/web/pages/submissions/upload.tsx [hunk 6]
```diff
@@ -106,12 +107,11 @@ const Upload = () => {
   const getCurrentSession = async () => {
     const session = await getSession();
     if (session === null) {
-      setUserSession(undefined);
+      router.push('/auth/login');
```
Comment (Huskydog9988) [inline apps/web/pages/submissions/upload.tsx:110]: why not just use `signIn`?

File: apps/web/server/trpc/router/gameplay.ts [hunk 7]
```diff
@@ -98,16 +98,17 @@ export const gameplayRouter = router({
...
     .output(GameplaySchema.array())
     .query(async ({ input, ctx }) => {
       // if no user id provided, use user id from session
       // userId should only be passed by system admins, not avg users
       // TODO: check roles and prevent users from getting other users
-      const userId = input.userId === null ? ctx.session.user.id : input.userId;
-
+      const userId =
+        input.userId === null ? ctx.session.user?.id : input.userId;
+      console.log(userId);
```
Comment (Huskydog9988) [inline apps/web/server/trpc/router/gameplay.ts:111]:
```suggestion
```

File: apps/web/tsconfig.json [hunk 8]
```diff
@@ -5,7 +5,8 @@
     "**/*.ts",
     "**/*.tsx",
     ".next/types/**/*.ts",
-    "pages/api/auth/[...nextauth].js"
+    "pages/api/auth/[...nextauth].js",
+    "pages/account.jsx"
```
Comment (Huskydog9988) [inline apps/web/tsconfig.json:9]: ?

File: apps/web/utils/helpers/apiHelper.ts [hunk 9]
```diff
@@ -42,4 +41,23 @@ const checkURL = (url: string): boolean => {
...
+  };
+  const url = baseUrl + videoId + '&key=' + `${process.env.YOUTUBE_API_KEY}`;
+  try {
+    const request = await fetch(url, options);
+    const res = await request.json();
+    return res.items[0].snippet;
+  } catch (error) {
+    throw new Error('Error Fetching');
+  }
+};
```
Comment (Huskydog9988) [inline apps/web/utils/helpers/apiHelper.ts:61]: (discussion on discord)

File: package.json [hunk 10]
```diff
@@ -29,9 +29,11 @@
   },
   "packageManager": "pnpm@7.15.0",
   "dependencies": {
+    "@chakra-ui/layout": "^2.1.11",
     "@chakra-ui/provider": "^2.0.25",
     "@heroicons/react": "^2.0.13",
     "async-ratelimiter": "^1.3.5",
+    "dotenv": "^16.0.3",
```
Comment (Huskydog9988) [inline package.json:36]: ?

**Review by Huskydog9988 — APPROVED** [review]

**Review by tsundDEV — APPROVED** [review]

**Pull request merged**